django_x509/base/models.py

```diff
@@ -349,48 +360,92 @@ def _import(self):
-        self.city = subject.localityName or ""
-        self.organization_name = subject.organizationName or ""
-        self.organizational_unit_name = subject.organizationalUnitName or ""
-        self.email = subject.emailAddress or ""
-        self.common_name = subject.commonName or ""
-        self.serial_number = cert.get_serial_number()
+        self.state = get_val(NameOID.STATE_OR_PROVINCE_NAME)
+        self.city = get_val(NameOID.LOCALITY_NAME)
+        self.organization_name = get_val(NameOID.ORGANIZATION_NAME)
+        self.organizational_unit_name = get_val(NameOID.ORGANIZATIONAL_UNIT_NAME)
+        self.common_name = get_val(NameOID.COMMON_NAME)
+        self.serial_number = str(cert.serial_number)
         if not self.name:
-            self.name = self.common_name or str(self.serial_number)
+            self.name = self.common_name or self.serial_number
 
     def _verify_ca(self):
         """
         (internal use only)
         verifies the current x509 is signed
         by the associated CA
         """
-        store = crypto.X509Store()
-        store.add_cert(self.ca.x509)
-        store_ctx = crypto.X509StoreContext(store, self.x509)
+        cert = self.x509
+        ca_cert = self.ca.x509
+        ca_pubkey = ca_cert.public_key()
+        if cert.issuer != ca_cert.subject:
+            raise ValidationError(
+                _("The Certificate Issuer does not match the CA Subject.")
+            )
+        now = timezone.now()
+        if cert.not_valid_before_utc > now or cert.not_valid_after_utc < now:
+            raise ValidationError(_("The certificate has expired or is not yet valid."))
+        try:
+            bc = ca_cert.extensions.get_extension_for_class(x509.BasicConstraints).value
+            if not bc.ca:
+                raise ValidationError(
+                    _(
+                        "The selected CA is not authorized to sign certificates"
+                        "(BasicConstraints)."
+                    )
+                )
+        except x509.ExtensionNotFound:
+            raise ValidationError(
+                _("The selected CA is missing BasicConstraints and cannot sign.")
+            )
         try:
-            store_ctx.verify_certificate()
-        except crypto.X509StoreContextError as e:
+            if isinstance(ca_pubkey, rsa.RSAPublicKey):
+                ca_pubkey.verify(
+                    cert.signature,
+                    cert.tbs_certificate_bytes,
+                    padding.PKCS1v15(),
+                    cert.signature_hash_algorithm,
+                )
+            elif isinstance(ca_pubkey, ec.EllipticCurvePublicKey):
+                ca_pubkey.verify(
+                    cert.signature,
+                    cert.tbs_certificate_bytes,
+                    ec.ECDSA(cert.signature_hash_algorithm),
+                )
+            else:
+                ca_pubkey.verify(cert.signature, cert.tbs_certificate_bytes)
+        except Exception:
```

Comment:
can we be more specific here with the exception being caught? This can catch other exceptions and report a wrong error message to the user.